frontend/src/App.tsx

```diff
@@ -1,24 +1,23 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import axios from "axios";
+import { useState } from "react";
+const serverURL = 'http://localhost:8000'
 
 function App() {
+  const [message, setMessage] = useState("");
+  
+  const handleClick = async () => {
+    if (message) {
+      setMessage("")
+      return
+    }
```

Comment:
i know this code is mostly just demonstrative and not going to be here for very long but it's a good idea in general to make your code as self documenting as possible. it took me a second to realize that you wanted to use this code block to "reset" the message being shown and you could've abstracted this out into a fn called `resetMessage` or similar.

that being said, this code actually isn't super useful, since your `setMessage` call on line 15 would effectively change the message being shown. in this case, you are actually causing react to rerender the dependent child components twice, once to clear it to "", and then shortly followed by the message passed from the server. obviously in this case, the performance drop isn't noticeable but this stuff will make a difference once you're coding at scale!

Reply:
I'm kind of confuse about why react would rerender the dependent chid component twice. I have the return statement after the setMessage(""). Can you elaborate on this? Also here's how I made the code more readable with a `resetMessage` fn:

      const handleClick = async () => {
      const displayingMessage = message ? true : false
       if (displayingMessage) {
            resetMessage()
            return 
      }
      ...

Reply:
oh sorry you're right! i glossed over it too quickly. having watched the video again, I realize now that the button operates as a reset button between API calls so this makes more sense now.

it looks good now!